### PR TITLE
Restructure into use cases and technology (repo unfrozen if merged)

### PR DESCRIPTION
**Pull request by ge0ffrey**
This is probably too big to decently review, even diagonally. But the changes are all related to the restructure

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix: {
-        quickstartDir: ['hello-world-school-timetabling', 'quarkus-school-timetabling', 'spring-boot-school-timetabling']
+        quickstartDir: ['hello-world', 'use-cases/school-timetabling', 'technology-integration/java-spring-boot']
```
Comment (rsynek) [inline .github/workflows/gradle.yml]:
```suggestion
        quickstartDir: ['hello-world', 'use-cases/school-timetabling', 'technology/java-spring-boot']
```

File: runQuickstartsFromSource.sh [hunk 2]
```diff
@@ -3,5 +3,7 @@
 # Change directory to the directory of the script
 cd "$(dirname $0)" || exit
 
-./mvnw verify -DskipTests &&
-  ./mvnw -f build/quickstarts-showcase quarkus:dev -Dstartup-open-browser=true
+cd build
+./mvnw -f .. verify -DskipTests &&
+  ./mvnw -f quickstarts-showcase quarkus:dev -Dstartup-open-browser=true
```
Comment (rsynek) [inline runQuickstartsFromSource.sh:8]: Running the showcase via this script (as well as via `mvn quarkus:dev`) seems to be broken; when I click on any of the quickstarts, I get:
/home/rsynek/projects/kie-group/optaplanner-quickstarts/build/mvnw: 219: cannot open /home/rsynek/projects/kie-group/optaplanner-quickstarts/use-cases/school-timetabling/.mvn/wrapper/maven-wrapper.properties: No such file
/home/rsynek/projects/kie-group/optaplanner-quickstarts/use-cases/school-timetabling/.mvn/wrapper/maven-wrapper.jar: No such file or directory
Error: Could not find or load main class org.apache.maven.wrapper.MavenWrapperMain

**Review by rsynek — CHANGES_REQUESTED** [review]
Looks like the Gradle CI and the showcase app are broken after the restructuring; see the comments inline.

File: runQuickstartsFromSource.sh (hunk 2 above)
Comment (ge0ffrey) [reply runQuickstartsFromSource.sh:8]: Strange, this worked for me. Mind if I merge this PR and fix this in another one separately?

**Review by rsynek — APPROVED** [review]

**Pull request merged**